GDWeave/Script/ModUtil.cs

```diff
@@ -10,6 +10,58 @@ public interface IWaiter {
     public bool Check(Token token);
 }
 
+public class FunctionWaiter : IWaiter {
+    private MultiTokenWaiter _FunctionWaiter = new([
+        t => t.Type == TokenType.PrFunction,
+        t => t.Type == TokenType.Identifier,
+    ]);
+
+    public bool Matched {get; private set;} = false;
+    public bool Ready {get; private set;} = true;
+
+    private readonly string name;
+    public FunctionWaiter(string name) {
+        this.name = name;
+    }
+
+    public void Reset() {
+        foundFunction = false;
+        foundColon = false;
+    }
+
+    public void SetReady() {
+        Ready = true;
+    }
+
+    private bool foundFunction = false;
+    private bool foundColon = false;
+    public bool Check(Token token) {
+        if (foundColon && token.Type == TokenType.Newline) {
+            Reset();
```

Comment:
This probably shouldn't reset instantly. I'll write better doc comments but the API surface is that `Matched` should be set to `true` on a match and then the user can manually `Reset` when needed.

Reply:
Done